### PR TITLE
(PXP-5524): feat/objects 

### PR DESCRIPTION
**Pull request by Avantol13**
### New Features
- new /objects API with support for a generic data upload flow into a "data lake" using fence/indexd behind the scenes

### Improvements
- capability of using tokens for authorization checks against arborist
- refactoring of config and support for new gino version
- new "global" async client (httpx.AsyncClient) for making requests via the FastAPI app (accessible in requests via "request.app.async_client")

### Dependency updates
- added respx for mocking httpx async requests for testing

### CODE REVIEW

File: src/mds/objects.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        logger.error(exc, exc_info=True)
+        raise HTTPException(
+            HTTP_401_UNAUTHORIZED,
+            f"Could not verify, parse, and/or validate scope from provided access token.",
+        )
+
+    file_name = body.dict().get("file_name")
+    authz = body.dict().get("authz")
+    aliases = body.dict().get("aliases") or []
+    metadata = body.dict().get("metadata")
```
Comment (fantix) [inline src/mds/objects.py:83]: just curious - wouldn't `body.file_nname` work?

File: src/mds/objects.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    }
+
+    return JSONResponse(response, HTTP_201_CREATED)
+
+
+async def _create_aliases_for_record(aliases: list, blank_guid: str, auth_header: str):
+    aliases_data = {"aliases": [{"value": alias} for alias in aliases]}
+    logger.debug(f"trying to create aliases: {aliases_data}")
+    try:
+        async with httpx.AsyncClient() as client:
```
Comment (fantix) [inline src/mds/objects.py:126]: sorry for this very late decision - we might want to sharer a single HTTP client instance across the application: https://www.python-httpx.org/advanced/#why-use-a-client

File: src/mds/objects.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        raise HTTPException(
+            HTTP_401_UNAUTHORIZED,
+            f"Could not verify, parse, and/or validate scope from provided access token.",
+        )
+
+    file_name = body.dict().get("file_name")
+    authz = body.dict().get("authz")
+    aliases = body.dict().get("aliases") or []
+    metadata = body.dict().get("metadata")
+    logger.debug(f"validating authz block input: {authz}")
```
Comment (fantix) [inline src/mds/objects.py:84]: (not a big deal) https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/

File: src/mds/objects.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    HTTP_403_FORBIDDEN,
+    HTTP_500_INTERNAL_SERVER_ERROR,
+)
+from pydantic import BaseModel
+
+from . import config, logger
+from .models import Metadata
+
+mod = APIRouter()
+arborist = ArboristClient()
```
Comment (fantix) [inline src/mds/objects.py:26]: (trivial) unused variable `arborist`

File: src/mds/objects.py (hunk 2 above)
Comment (Avantol13) [reply src/mds/objects.py:126]: ah, interesting, is it okay to leave the connection open as along as the application is running? are you thinking we setup the client during app creation like

```
async_client = httpx.AsyncClient()
```

and then just import and use it everywhere without an explicit `.close()`?

Comment (fantix) [reply src/mds/objects.py:124]: Right, I think it is okay to do that. To be clean, we may want to call `close()` on application shutdown.

File: src/mds/asgi.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+from . import logger
+
+app = get_app()
+
+
+@app.on_event("shutdown")
+def shutdown_event():
+    logger.info("Closing async client.")
+    global app
+    app.async_client.close()
```
Comment (fantix) [inline src/mds/asgi.py:11]: could we pls put this into `get_app()`?

Comment (Avantol13) [reply src/mds/asgi.py:11]: yep, will do

**Pull request merged**